docs/proposals/backup-restore-multi.md

```diff
@@ -0,0 +1,881 @@
+
+```bash
+oras repo ls --oci-layout-path base-images.tar
+```
+
+**Output:**
+```console
+docker.io/library/alpine
+docker.io/library/busybox
+docker.io/library/nginx
+```
+
+Or with a namespace filter:
+
+```bash
+oras repo ls --oci-layout-path base-images.tar docker.io/library
+```
+
+**Output:**
+```console
+alpine
+busybox
+nginx
+```
+
+### List Tags in a Repository
+
+```bash
+oras repo tags --oci-layout-path base-images.tar docker.io/library/busybox
+```
+
+**Output:**
+```console
+1.36.1
+```
+
+### Show Manifest Details
+
+```bash
+oras manifest fetch --oci-layout-path base-images.tar docker.io/library/busybox:1.36.1
+```
+
+### Discover Referrers
+
+```bash
+oras discover --oci-layout-path base-images.tar docker.io/library/busybox:1.36.1
+```
+
+**Note:** All ORAS commands supporting `--oci-layout-path` can inspect backup contents with repository paths.
+
+## Dependencies
+
+This proposal depends on the following:
+
+- **[#1770](https://github.com/oras-project/oras/issues/1770)**: Support `oras repo ls --oci-layout-path` for listing repositories in OCI Image Layout backups.
+
+## Backup Format and Annotations
+
+Multi-repository backups are stored in standard OCI Image Layout format with full reference information preserved in annotations.
+
+### Reference Annotation (`org.opencontainers.image.ref.name`)
+
+The `org.opencontainers.image.ref.name` annotation in `index.json` stores the artifact reference:
+
+- **Default behavior (backward compatible)**: Contains just the tag (e.g., `"latest"`)
+- **With `--full-reference` flag**: Contains the full reference (e.g., `"docker.io/library/alpine:latest"`)
+- **Multi-repository backup**: Always uses full reference regardless of flag
+
+#### The `--full-reference` Flag
+
+The `--full-reference` flag can be used with single-repository backups to store the complete reference path:
+
+```bash
+oras backup --output backup.tar --full-reference docker.io/library/alpine:latest
+```
+
+This approach:
+- Aligns with the [OCI Image Spec](https://github.com/opencontainers/image-spec/blob/v1.1.1/annotations.md#pre-defined-annotation-keys), which allows both tag-only and full reference formats in `org.opencontainers.image.ref.name`
+- Enables automation tools (e.g., air-gap packaging tools like [Zarf](https://github.com/zarf-dev/zarf)) to programmatically determine the original location of backed-up artifacts
+- Addresses use cases described in [#1893](https://github.com/oras-project/oras/issues/1893)
+- **Multi-repository backups always use full references** to disambiguate artifacts from different repositories
+
+> [!WARNING]
+> Backups created with `--full-reference` are **not compatible** with ORAS CLI versions prior to v1.4.0 or tools that expect tag-only references in `org.opencontainers.image.ref.name`. Use this flag only when you need full reference information for automation or multi-registry scenarios.
+
+**Example `index.json` formats:**
+
+**Default single-repository backup (tag-only):**
```

Comment:
Seems like it would be better to make full-reference default and something else to make these short references.

We haven't had a release out yet with these short references yet, have we? Not to mention this feature is experimental

Reply:
I think the short references (tags) are generated in the image annotations since `oras v1.3.0`.

```
"annotations":{"org.opencontainers.image.ref.name":"v1.2.3-2"}
```

Reply:
I think there is a backward compatible issue, not just for current implementation of `oras backup`/`oras restore` but `oras repo tags` and other command like `oras copy`, if we use full-reference by default for single repo.